### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jeyldii**

### CODE REVIEW

File: controllers/getcoursescontroller.go [hunk 1]
```diff
@@ -31,23 +31,17 @@ type dataTokensAndCurrencies struct {
...
-
-type percentChanges struct {
-	Rates map[string]string
-	PercentChange string `json:"percent_change,omitempty"`
+	Currency string              `json:"currency"`
+	Rates    []map[string]string `json:"rates"`
 }
 
 // make Response list API
 type listApi struct {
```
Comment (krboktv) [inline controllers/getcoursescontroller.go]: Is there a method to this structure? If not, the use of structure is unnecessary

File: controllers/getcoursescontroller.go [hunk 2]
```diff
@@ -141,14 +134,38 @@ func (cr *controller) converterCMCW(req *dataTokensAndCurrencies) (*[]prices, er
...
+		}
+
+		return result, nil
+
+	default:
+		return nil, errors.New("no matches API changes")
+	}
+}
+
+func (cr *controller) converterCRC(req *dataTokensAndCurrencies) ([]prices, error) {
```
Comment (krboktv) [inline controllers/getcoursescontroller.go]: Think about splitting this function into several. Too big

File: storage/storecrc/store.go [hunk 3]
```diff
@@ -13,34 +12,33 @@ type Currency struct {
...
+func (r *storedList) Update(res map[string]Fiat) {
 	r.mu.Lock()
-	defer r.mu.Unlock()
 	r.Stored = res
+	r.mu.Unlock()
 }
 
-func (r *storedList) Get() map[string]Cr {
+func (r *storedList) Get() map[string]Fiat {
 	r.mu.Lock()
```
Comment (krboktv) [inline storage/storecrc/store.go]: Lock => RLock

File: storage/storetrustwallet/store.go [hunk 4]
```diff
@@ -22,25 +22,25 @@ type DocsPrices struct {
...
+func (r *storedPrices) Update(res []*GotPrices) {
 	r.mu.Lock()
-	defer r.mu.Unlock()
 	r.Stored = res
+	r.mu.Unlock()
 }
 
-func (r *storedPrices) Get() []GotPrices {
+func (r *storedPrices) Get() []*GotPrices {
 	r.mu.Lock()
```
Comment (krboktv) [inline storage/storetrustwallet/store.go]: Lock =>RLock

File: slip0044/slips.go [hunk 5]
```diff
@@ -1,39 +1,70 @@
...
+			Contract:   slipHexed,
+			CoinSymbol: constants[i].CoinSymbol,
+		}
+		trustWalletSlips = append(trustWalletSlips, &trustWalletSlip)
+	}
+
+	return trustWalletSlips, nil
+}
+
+func makeHexString(s string) (string, error) {
```
Comment (krboktv) [inline slip0044/slips.go:37]: Look at `hex` go package

File: storage/storetrustwallet/store.go (hunk 4 above)
Comment (prazd) [reply storage/storetrustwallet/store.go]: ++++!!

File: storage/storecrc/store.go (hunk 3 above)
Comment (prazd) [reply storage/storecrc/store.go]: +++++!!

File: controllers/getcoursescontroller.go (hunk 1 above)
Comment (prazd) [reply controllers/getcoursescontroller.go]: Are you sure?)

**Review by prazd — CHANGES_REQUESTED** [review]
Many things need to be simplified

**Comment by EnoRage** [comment]
https://www.ozon.ru/context/detail/id/141508653/

**Comment by EnoRage** [comment]
Needs additional fixes

**Pull request merged**